### PR TITLE
Vedligehold2020

### PR DESCRIPTION
**Pull request by thomasegense**
Complete review of refactoring architecture team guideline

### CODE REVIEW

File: changelog.txt [hunk 1]
```diff
@@ -1,3 +1,5 @@
+v.4.0:bla bla
```
Comment (sbktc) [inline changelog.txt:1]: bla bla? I think this could be improved somewhat

File: pom.xml [hunk 2]
```diff
@@ -1,298 +1,244 @@
...
+			<groupId>com.h2database</groupId>
+			<artifactId>h2</artifactId>
+			<version>1.3.170</version>
+			<scope>test</scope>
+		</dependency>
+
+		<dependency>
+			<groupId>org.postgresql</groupId>
+			<artifactId>postgresql</artifactId>
+			<version>9.4.1208</version>
```
Comment (sbktc) [inline pom.xml:64]: Postgres apparently have changed their versioning of their maven dependencies. 9.4.x is an older version, consider updating to the latest 42.2.14 version

File: pom.xml [hunk 3]
```diff
@@ -1,298 +1,244 @@
...
+			<groupId>org.apache.commons</groupId>
+			<artifactId>commons-dbcp2</artifactId>
+			<version>2.0</version>
+		</dependency>
+
+
+		<dependency>
+			<groupId>junit</groupId>
+			<artifactId>junit</artifactId>
+			<version>4.8.2</version>
```
Comment (sbktc) [inline pom.xml:78]: Please update junit framework to junit5 (jupiter)

File: src/main/java/dk/statsbiblioteket/doms/licensemodule/service/LicenseModuleResource.java [hunk 4]
```diff
@@ -56,8 +56,8 @@ public CheckAccessForIdsOutputDTO checkAccessForIdsJSON(CheckAccessForIdsInputDT
 	
 	
 	@POST	
-	@Consumes(MediaType.TEXT_XML)	
-	@Path("checkAccessForIds")	
+	@Path("checkAccessForIds")
+	@Consumes(MediaType.TEXT_XML)			
 	@Produces(MediaType.TEXT_XML)
```
Comment (sbktc) [inline src/main/java/dk/statsbiblioteket/doms/licensemodule/service/LicenseModuleResource.java:61]: As Asger noted, consider adding MediaType.APPLICATION_XML to the list of acceptable Consumes and Produces

File: src/main/java/dk/statsbiblioteket/doms/licensemodule/service/LicenseModuleResource.java [hunk 5]
```diff
@@ -104,21 +104,11 @@ public ValidateAccessOutputDTO validateAccess(ValidateAccessInputDTO input)
...
-	    log.info("getUserLicensesJSON called");
-		return getUserLicenses(input);
-	}		
-	
-	@POST	
-	@Path("getUserLicenses")	
-	@Consumes(MediaType.TEXT_XML)	
-	@Produces(MediaType.TEXT_XML)
+	@Consumes({MediaType.TEXT_XML,MediaType.APPLICATION_JSON})	
+	@Produces({MediaType.TEXT_XML,MediaType.APPLICATION_JSON})
```
Comment (sbktc) [inline src/main/java/dk/statsbiblioteket/doms/licensemodule/service/LicenseModuleResource.java:111]: Same as the comment pertaining to line 59 and 60

File: src/test/jetty/jetty-env.xml [hunk 6]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0"?>
+<Configure class="org.eclipse.jetty.webapp.WebAppContext">
+
+    <New id="logback" class="org.eclipse.jetty.plus.jndi.EnvEntry">
+        <Arg>java-webapp-logback-config</Arg>
```
Comment (sbktc) [inline src/test/jetty/jetty-env.xml:5]: You will want the value to match the name of the log configuration. i.e.: licensemodule-logback-config

File: pom.xml [hunk 7]
```diff
@@ -1,298 +1,244 @@
...
+			<artifactId>logback-classic</artifactId>
+			<version>1.2.3</version>
+		</dependency>
+
+
+		<!-- SOLRJ -->
+		<dependency>
+			<groupId>org.apache.solr</groupId>
+			<artifactId>solr-solrj</artifactId>
+			<version>4.0.0</version>
```
Comment (sbktc) [inline pom.xml:155]: Ancient version, consider updating to a more current? 8.5.2 perhaps

Comment (thomasegense) [reply pom.xml]: upgraded to 8.5.2 (minor code rewrite)

File: pom.xml (hunk 2 above)
Comment (thomasegense) [reply pom.xml]: Upgraded to 42.2.14 (and tested)

File: src/test/jetty/jetty-env.xml (hunk 6 above)
Comment (thomasegense) [reply src/test/jetty/jetty-env.xml]: Fixed, also added the property file as env variable. Added template property to jetty folder

File: pom.xml (hunk 3 above)
Comment (thomasegense) [reply pom.xml]: Done. Minor rewrite of test-code

File: src/main/java/dk/statsbiblioteket/doms/licensemodule/service/LicenseModuleResource.java (hunk 5 above)
Comment (thomasegense) [reply src/main/java/dk/statsbiblioteket/doms/licensemodule/service/LicenseModuleResource.java]: Done

File: src/main/java/dk/statsbiblioteket/doms/licensemodule/service/LicenseModuleResource.java (hunk 4 above)
Comment (thomasegense) [reply src/main/java/dk/statsbiblioteket/doms/licensemodule/service/LicenseModuleResource.java]: Done

File: changelog.txt (hunk 1 above)
Comment (thomasegense) [reply changelog.txt]: More details now...

**Pull request merged**